pandas/core/indexes/multi.py

```diff
@@ -2655,8 +2656,9 @@ def equals(self, other):
 
             olabels = other.labels[i]
             olabels = olabels[olabels != -1]
-            ovalues = algos.take_nd(np.asarray(other.levels[i]._values),
-                                    olabels, allow_fill=False)
+            ovalues = algos.take_nd(
+                np.asarray(other.levels[i]._values),
```

Comment:
why not _ndarray_values here?

Reply:
Breaks for categoricals. We've baked knowledge of cats into `algost.take_nd` / `take_1d`, to handle it.